src/scheduler/Executor.cpp

```diff
@@ -588,6 +588,9 @@ void Executor::threadPoolThread(std::stop_token st, int threadPoolIdx)
             releaseClaim();
         }
 
+        // Clear the thread local PTP cache
```

Comment:
I am clearing the thread local cache every time an execuing thread is done executing a task. This feels like the right choice for non-threads, but it may not be for thread execution (?).

Reply:
I don't think this is a good idea, each executor is either repeatedly executing jobs from the same application (in which case the cache is useful and we want it to persist), or will automatically get cleared up by the scheduler reaping process, or will be cleared from an explicit flush operation. Its associated TLS will automatically get cleaned up when the underlying thread is removed in the last two cases.

Unless it's definitely needed to avoid resource exhaustion, I don't see what benefit this adds, and think it's sufficient just to add the clean-up to the scheduler. I would be surprised if it was this TLS that's the problem (i.e. I suspect it's the stuff that's outside the executor lifecycle that builds up but may be wrong).